kinto/core/storage/memory.py

```diff
@@ -329,13 +329,14 @@ def apply_filters(records, filters):
             if f.operator in (COMPARISON.IN, COMPARISON.EXCLUDE):
                 right, left = left, right
             else:
-                # Python3 cannot compare None to other value.
                 if left is None:
-                    if f.operator in (COMPARISON.GT, COMPARISON.MIN):
+                    if isinstance(right, (int, float)) and \
+                       right not in (True, False):
```

Comment:
nitpicks: use intermediary variables to avoid folding if conditions (ex. `is_number = (....)`)